universal-application-tool-0.0.1/app/views/BaseHtmlLayout.java

```diff
@@ -60,4 +69,88 @@ public String contentType() {
+                .withClasses(
+                    Styles.MAX_W_7XL,
+                    Styles.MX_AUTO,
+                    Styles.PY_3,
+                    Styles.PX_3,
+                    StyleUtils.responsiveSmall(Styles.PX_6),
+                    StyleUtils.responsiveLarge(Styles.PX_8))
+                .with(
+                    div()
+                        .withClasses(
+                            Styles.FLEX,
+                            Styles.ITEMS_CENTER,
+                            Styles.JUSTIFY_BETWEEN,
+                            Styles.FLEX_WRAP)
+                        .with(
+                            div()
+                                .withClasses(
+                                    Styles.W_0, Styles.FLEX_1, Styles.FLEX, Styles.ITEMS_CENTER)
+                                .with(
+                                    span()
+                                        .withClasses(
+                                            addBgOpacity(
+                                                Styles.FLEX,
+                                                Styles.P_2,
+                                                Styles.ROUNDED_LG,
+                                                Styles.BG_RED_800))
+                                        .with(
+                                            Icons.svg(Icons.LOGIN_BANNER_PATH, 24)
+                                                .withClasses(
+                                                    Styles.H_6, Styles.W_6, Styles.TEXT_WHITE)
+                                                .attr("fill", "none")
+                                                .attr("stroke-linecap", "round")
+                                                .attr("stroke-linejoin", "round")
+                                                .attr("stroke-width", "2")),
+                                    p().withClasses(
+                                            Styles.ML_3,
+                                            Styles.FONT_MEDIUM,
+                                            Styles.TEXT_WHITE,
+                                            Styles.TRUNCATE)
+                                        .with(span(BANNER_TEXT))))));
+  }
+
+  private String[] addBgOpacity(String... styles) {
+    String[] newStyles = Arrays.copyOf(styles, styles.length + 1);
+    newStyles[styles.length] = Styles.BG_OPACITY_80;
+    return newStyles;
+  }
+
+  private boolean isAdminPage() {
+    return this.getClass().getCanonicalName().startsWith("views.admin");
+  }
+
+  private String[] demoBannerStyles() {
+    if (isAdminPage()) {
+      return addBgOpacity(Styles.BG_RED_600, Styles.ABSOLUTE, Styles.BOTTOM_2, Styles.W_FULL);
+    }
+    return new String[] {Styles.BG_RED_600};
+  }
+
+  private DomContent[] addDemoBanner(DomContent... domContents) {
+    if (isAdminPage()) {
+      return appendDomContent(domContents, demoBanner());
+    }
+    return prependDomContent(domContents, demoBanner());
```

Comment:
Thoughts: this has an unfortunate side-effect of showing an un-styled banner before the tailwind styles are loaded. We probably want to move to appending all banners instead of prepending them.

Reply:
Let's take another pass at this. I don't love the branching logic, so I think we could probably simplify things if we come up with a style that we can use across the board.

Reply:
Yeah, that looks way better. thanks!